pyez/pyez_building_blocks/bb1.collecting.and.parsing.configuration.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python
+#
+# Copyright 2017 Juniper Networks, Inc. All rights reserved.
+# Licensed under the Juniper Networks Script Software License (the "License").
+# You may not use this script file except in compliance with the License, which is located at
+# http://www.juniper.net/support/legal/scriptlicense/
+# Unless required by applicable law or otherwise agreed to in writing by the parties, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied.
+#
+# Author.........: Diogo Montagner <dmontagner@juniper.net>
+# Created on.....: 15/Dec/2017
+# Version........: 1.0
+# Platform.......: agnostic
+# Description....: Simple example of configuration collection and parsing
+#
+
+
+import logging
+import sys
+import datetime
+import pprint
+from jnpr.junos import Device
+from lxml import etree
+from collections import defaultdict
+from netaddr import *
+
+
+# setting logging capabilities
+log = logging.getLogger() # 'root' Logger
+console = logging.StreamHandler()
+format_str = '%(asctime)s\t%(levelname)s -- %(processName)s %(filename)s:%(lineno)s -- %(message)s'
+console.setFormatter(logging.Formatter(format_str))
+log.addHandler(console) # prints to console.
+
+# set the log level here
+#log.setLevel(logging.WARN) 
+log.setLevel(logging.ERROR) 
+
+
+#
+# This method collects the configuration from the router
+#
+# Returns the the filename where the configuration was stored
+#
+def getConfigurationFromRouter(userName, userPassword, router, format):
+
+    log.debug("entered getConfigurationFromRouter")
+    cnf = None
+    FileName = None
+
+    try:
+        log.debug("host = %s, user = %s, password = %s, router = %s, format = %s", userName, userPassword, router, format)
+        dev = Device(host=router, user=userName, password=userPassword, gather_facts=False)
+        dev.open()
+
+    except Exception as e:
+        log.error("could not connect to the router %s", router)
+        log.error(e.message)
+        exit(-1)
+
+    try:
+        log.debug("collecting the router configuration")
+        
+        now = datetime.datetime.now()
+        datets = str(now.year) + str(now.month) + str(now.day) + "_" + str(now.hour) + str(now.minute) + str(now.second)
+
+        if (format == "cnf"):
+            cnf = dev.cli("show configuration", warning=False)
+            FileName = router + "." + datets + ".cnf"
+            log.warn("The configuration will be stored in filename as %s", FileName)
+
+            # saving the configuration into a CNF file
+            f = open(FileName, 'w+')
+            f.write(cnf)
+            f.close
+            return FileName
```

Comment:
above 2 conditions is a repetition, how about something like below
format is python keyword, check if we can change it to some other name (i used ext)
if ext in ["set", "set"]:
    FileName = router + "." + datets + "." + ext
   cmd = "show configuration"
    if ext == "set":
        cmd = cmd + " | display set"
    cnf = dev.cli(cmd, warning=False)
    log.warn("The configuration will be stored in filename as %s", FileName)
    # saving the configuration into a file
    with open(FileName, 'w+') as fp:
        fp.write(cnf)
    return FileName